core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -294,9 +294,14 @@ private[spark] object Utils extends Logging {
       }
       try {
         dir = new File(root, namePrefix + "-" + UUID.randomUUID.toString)
-        if (dir.exists() || !dir.mkdirs()) {
```

Comment:
I think we intendedly try to return a non-existent and a newly created directory. It looks not a hot path and performance here looks not a big deal.